### DISCUSSION TITLE
How can I  get the data type of the cell value such as number, percentage, string

### DISCUSSION THREAD

**Discussion by bvpranu97** [exceljs/exceljs#1963] [Q&A]
I want to get the data type of a cell to validate it how can I do? If the cell has value ending with % then it is considered as percent and it is / by 100 that value is inconsistant because Iam not calculating anything  from the modified value.

**Comment by skypesky** [comment]
use `typeof`
```typescript
if(typeof value === 'number') {
  // do something
} else if(typeof value === 'string') {
  if(value.endsWith('%')) {
    // ending with %
  } else {
  
  }
}
```

use `cell.type`

```typescript
 const cell: Cell = worksheet.getCell('A1');
 console.info(cell.type === ValueType.Number);
```

**Comment by skypesky** [comment]
@bvpranu97 
Has your problem been solved?
